crates/sol-macro-gen/src/sol_macro_gen.rs

```diff
@@ -0,0 +1,367 @@
+        };
+        if !single_file {
+            for instance in &self.instances {
+                let name = instance.name.to_lowercase();
+                let path = crate_path.join(format!("src/{}.rs", name));
+                let tokens = instance
+                    .expansion
+                    .as_ref()
+                    .ok_or_eyre(format!("TokenStream for {path:?} does not exist"))?
+                    .to_string();
+
+                self.check_file_contents(&path, &tokens)?;
+
+                if !is_mod {
+                    super_contents += &format!(
+                        r#"pub mod {};
+                    "#,
+                        name
+                    );
+                }
+            }
+
+            let super_path =
+                if is_mod { crate_path.join("src/mod.rs") } else { crate_path.join("src/lib.rs") };
+            self.check_file_contents(&super_path, &super_contents)?;
+        }
+
+        Ok(())
+    }
+
+    fn check_file_contents(&self, file_path: &Path, expected_contents: &str) -> Result<()> {
+        eyre::ensure!(
+            file_path.is_file() && file_path.exists(),
+            "{} is not a file",
+            file_path.display()
+        );
+        let file_contents =
+            &fs::read_to_string(file_path).map_err(|e| eyre::eyre!("Failed to read file: {e}"))?;
+        eyre::ensure!(
+            file_contents == expected_contents,
+            "File contents do not match expected contents for {file_path:?}"
+        );
+        Ok(())
+    }
+
+    fn check_cargo_toml(
+        &self,
+        name: &str,
+        version: &str,
+        crate_path: &Path,
+        alloy_version: String,
+    ) -> Result<()> {
+        eyre::ensure!(crate_path.is_dir(), "Crate path must be a directory");
+
+        let cargo_toml_path = crate_path.join("Cargo.toml");
+
+        eyre::ensure!(cargo_toml_path.is_file(), "Cargo.toml must exist");
+        let cargo_toml_contents = fs::read_to_string(cargo_toml_path)
+            .map_err(|e| eyre::eyre!("Failed to read Cargo.toml: {e}"))?;
+
+        let name_check = &format!("name = \"{}\"", name);
+        let version_check = &format!("version = \"{}\"", version);
+        let sol_types_check = &format!("alloy-sol-types = \"{}\"", alloy_version);
+        let alloy_contract_check =
+            "alloy-contract = {{ git = \"https://github.com/alloy-rs/alloy\", rev = \"64feb9b\" }}";
+        let toml_consistent = cargo_toml_contents.contains(name_check) &&
+            cargo_toml_contents.contains(version_check) &&
+            cargo_toml_contents.contains(sol_types_check) &&
+            cargo_toml_contents.contains(alloy_contract_check);
+        eyre::ensure!(
+                toml_consistent,
+                format!("The contents of Cargo.toml do not match the expected output of the newest `ethers::Abigen` version.\n
+This indicates that the existing bindings are outdated and need to be generated again.")
```

Comment:
rustfmt bugs out on long literals, pls remove `format!` and put it on multiple lines; also it's mentioning ethers abigen